contracts/ocr2/src/contract.rs

```diff
@@ -259,6 +258,32 @@ pub fn reply(_deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, Contract
     }
 }
 
+// Migrate contract if version is lower than current version
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, _env: Env, _msg: Empty) -> Result<Response, ContractError> {
+    let ver = cw2::get_contract_version(deps.storage)?;
+
+    // ensure we are migrating from an allowed contract
+    if ver.contract != CONTRACT_NAME {
+        return Err(cosmwasm_std::StdError::generic_err("Can only upgrade from same type").into());
```

Comment:
Why don't we explicitly define a `MigrationTypeMismatch` error in `error.rs`.

Reply:
Doesn't really matter since we're not catching the error anywhere, it's just used in `tx migrate`. Can change it though

Reply:
I have a slight preference to change it, as it makes the error more explicit and discoverable. At the same time code is more readable (consider tests) and typed.

Additionally, I guess at one point we could add a base crate and share some of these base types and functionality like we currently do with the `Owned` crate (sharing the `Unauthorized` error in this context).

It is a nitpick ATM though, so feel free to go either direction for now.